core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -1199,7 +1199,7 @@ public void removeListener(MuleContextListener listener) {
   private static void configureSplitterExceptionHandlingFeature() {
     FeatureFlaggingRegistry featureFlaggingRegistry = FeatureFlaggingRegistry.getInstance();
     featureFlaggingRegistry.registerFeatureFlag(HANDLE_SPLITTER_EXCEPTION, featureContext -> featureContext
-        .getArtifactMinMuleVersion().filter(muleVersion -> muleVersion.atLeast("4.4.0")).isPresent());
+        .getArtifactMinMuleVersion().filter(muleVersion -> muleVersion.withoutSuffixes().atLeast("4.4.0")).isPresent());
```

Comment:
if the idea is that mule apps follow semantic versioning, then the muleVerison received in these callbacks must already be without suffixes. Otherwise you need to repeat the call to `withoutSuffixes()` on every usage.